packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,329 @@
+        return this;
+    }
+    public get _root() {
+        return this.root;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+
+    private readonly matrixKey = "SharedMatrix";
+    private readonly counterKey = "Counter";
+    public matrix!: SharedMatrix;
+    public undoRedoStackManager!: UndoRedoStackManager;
+    public counter!: SharedCounter;
+
+    protected async initializingFirstTime() {
+        const sharedMatrix = SharedMatrix.create(this.runtime, this.matrixKey);
+        this.root.set(this.matrixKey, sharedMatrix.handle);
+
+       const counter = SharedCounter.create(this.runtime, this.counterKey);
+       this.root.set(this.counterKey, counter.handle);
+
+       const dsFactory2 = await requestFluidObject<TestDataObject2>(
+       await this._context.containerRuntime.createDataStore(TestDataObjectType2), "");
+       this.root.set("dsFactory2", dsFactory2.handle);
+    }
+
+    protected async hasInitialized() {
+        const matrixHandle = this.root.get<IFluidHandle<SharedMatrix>>(this.matrixKey);
+        assert(matrixHandle !== undefined, "SharedMatrix not found");
+        this.matrix = await matrixHandle.get();
+
+        this.undoRedoStackManager = new UndoRedoStackManager();
+        this.matrix.insertRows(0, 3);
+        this.matrix.insertCols(0, 3);
+        this.matrix.openUndo(this.undoRedoStackManager);
+
+        const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(this.counterKey);
+        assert(counterHandle);
+        this.counter = await counterHandle.get();
+    }
+}
+const dataStoreFactory1 = new DataObjectFactory(
+    TestDataObjectType1,
+    TestDataObject1,
+    [SharedMap.getFactory(), SharedMatrix.getFactory(), SharedCounter.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const dataStoreFactory2 = new DataObjectFactory(
+    TestDataObjectType2,
+    TestDataObject2,
+    [SharedMap.getFactory(), SharedMatrix.getFactory(), SharedCounter.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+    runtime.IFluidHandleContext.resolveHandle(request);
+
+const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+    [dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+    [dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+    ],
+);
+
+const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+    dataStoreFactory1,
+    registryStoreEntries,
+    undefined,
+    [innerRequestHandler],
+    runtimeOptions,
+);
+
+async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+): Promise<ISummarizer> {
+    return createSummarizerFromFactory(
+        provider,
+        container,
+        dataStoreFactory1,
+        summaryVersion,
+        getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore,
```

Comment:
You can directly pass `runtimeFactory` here and the next parameter is not needed.